src/components/typeahead/typeahead.component.ts

```diff
@@ -191,6 +215,22 @@ export class TypeaheadComponent<T = any> implements OnChanges, OnDestroy {
 
         // Re-filter visibleOptions
         this.updateOptions();
+
+        if (changes.recentOptions) {

```

Comment:
Could we make `recentOptions` a setter instead of using `ngOnChanges`, as ngOnChanges will only get triggered when change detection runs, but if I was to set this array programmatically the observable would not get updated, eg:

```
@ViewChild(SelectComponent, { static: false ) select: SelectComponent;

...
ngAfterViewInit(): void {
    this.select.recentOptions = [...];
}
```

Whereas with a setter this would update the observable

Reply:
The only time we change the recent options programmatically is in the optionClickHandler. We are using array functions there, though, like unshift() and splice(), which do not call the setter function. Therefore, we still need to call the update function for the recent options observable.

So, at the moment, it does not gain much, but adds complexity. You are right that if someone changes the code later, they will be able to reassign the recent options without taking care of the observable. If they manipulate the array, again, in a different way, it will not help.

I would therefore like to keep it as it is. Please tell me your opinion.

Reply:
The problem you mention about the setter not being called by `unshift` and `splice`, this is another reason for using `ReadonlyArray`. If using a `ReadonlyArray` then we must create a new array with the updated content each time we change it resulting in the setter being called every time we change it.

In fact by changing the array to be readonly we may be able to get rid of the observable all together

Reply:
Done